src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -0,0 +1,276 @@
+
+-/
+
+noncomputable theory
+open measure_theory
+open set
+open filter
+open_locale topological_space ennreal nnreal
+
+section finite_measures
+/-!
+### Finite measures
+In this section, we define the `Type` of finite measures on a given measurable space. This type
+will be equipped with the topology of weak convergence of measures when the measurable space is
+a topological space equipped with its Borel sigma-algebra.
+-/
+
+variables {α : Type} [measurable_space α]
+
+/-- Finite measures are defined as the subtype of measures that have the property of being finite
+measures (i.e., their total mass is finite). -/
+def finite_measures (α : Type*) [measurable_space α] : Type :=
+{ μ : measure α // finite_measure μ }
+
+namespace finite_measures
+
+instance has_zero {α : Type*} [measurable_space α] :
+  has_zero (finite_measures α) :=
+⟨{ val := 0,
+   property := measure_theory.finite_measure_zero, }⟩
+
+instance : inhabited (finite_measures α) := { default := 0 }
+
+instance {α : Type*} [measurable_space α] : has_add (finite_measures α) :=
+{ add := (λ (μ ν : finite_measures α),
+  { val := μ.val + ν.val,
+    property := @measure_theory.finite_measure_add α _ μ.val ν.val μ.prop ν.prop, }), }
+
+instance {α : Type*} [measurable_space α] : has_scalar ℝ≥0 (finite_measures α) :=
+{ smul := (λ (c : ℝ≥0) (μ : finite_measures α),
+  { val := c • μ.val,
+    property := @measure_theory.finite_measure_smul_nnreal α _ μ.val μ.prop c, }), }
+
+-- TODO: A coercion `finite_measures α → measure α` is crucial. Now it is the subtype coercion.
+-- Another alternative is to use a bijection with `vector_measure α ℝ≥0` as an intermediate step.
+-- Potential advantages of the vector measure alternative:
+--  * Structures of additive commutative monoid and ℝ≥0 module on `finite_measure α` could be
+--    "inherited".
+-- Potential drawbacks of the vector measure alternative:
+--  * No integration theory directly.
+--  * Monotonicity lost as non-measurable sets defined to have measure 0.
+
+/-- A finite measure can be interpreted as a measure. -/
+instance : has_coe (finite_measures α) (measure_theory.measure α) := coe_subtype
+
+instance to_measure.finite_measure (μ : finite_measures α) :
+  finite_measure (μ : measure α) := μ.prop
+
+instance (α : Type*) [measurable_space α] :
+  has_coe_to_fun (finite_measures α) := ⟨(λ _, set α → ℝ≥0),
+    (λ μ, (λ s, (μ.val.measure_of s).to_nnreal))⟩
+
+lemma to_fun_eq_to_measure_to_nnreal (ν : finite_measures α) :
+  (ν : set α → ℝ≥0) = λ s, ((ν : measure α) s).to_nnreal := rfl
+
+lemma to_measure_eq_val (ν : finite_measures α) : (ν : measure α) = ν.val := rfl
+
+lemma coe_injective :
+  function.injective (coe : finite_measures α → measure α) :=
+subtype.coe_injective
+
+@[simp]
+lemma coe_zero : (coe : finite_measures α → measure α) 0 = 0 := rfl
+
+@[simp]
+lemma coe_add (μ ν : finite_measures α) :
+  (coe : finite_measures α → measure α) (μ + ν) = (μ : measure α) + (ν : measure α) := rfl
```

Comment:
This is shorter to write (but otherwise identical) as:
```suggestion
@[simp] lemma coe_add (μ ν : finite_measures α) : ↑(μ + ν) = (↑μ + ↑ν : measure α) := rfl
```

Reply:
Ok, thank you!

I was quite surprised that this syntax works, though. With parentheses and _explicitly_ telling the intended coerced type, Lean interpreted the coercion to take place before the addition (the two sides became syntactically equal, I guess). It is good to know that with `↑` this can be circumvented :astonished:.

Reply:
It makes a lot more sense if you read `(x + y : Z)` as "lean, interpret the expression `x + y` as a `Z`" and not "lean, coerce the _value_ `x + y` to a `Z`".